### ISSUE TITLE
Having non-converted operations, even for simplest models

### ISSUE THREAD

**Issue by gaikwadrahul8** [google-ai-edge/LiteRT#84]
### System information

- Platform: Tried on Google Colab
- TensorFlow version: 2.15.0

### Steps to reproduce

- Creating a Python file with the following content in Google Colab (let's call it test.py):
```Python
import tensorflow as tf

model = tf.keras.models.Sequential([
    tf.keras.layers.InputLayer(input_shape=(1,)),
    tf.keras.layers.Dense(1)
])

converter = tf.lite.TFLiteConverter.from_keras_model(model)
tflite_model = converter.convert()

with open('model.tflite', 'wb') as f:
  f.write(tflite_model)
```

- Calling it in Jupyter Notebook by `!python test.py`
- The output will be:

```Python
2024-01-28 11:17:11.939381: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-01-28 11:17:11.939450: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-01-28 11:17:11.941203: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-01-28 11:17:13.646897: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2024-01-28 11:17:16.713671: W tensorflow/compiler/mlir/lite/python/tf_tfl_flatbuffer_helpers.cc:378] Ignored output_format.
2024-01-28 11:17:16.713736: W tensorflow/compiler/mlir/lite/python/tf_tfl_flatbuffer_helpers.cc:381] Ignored drop_control_dependency.
Summary on the non-converted ops:
---------------------------------
 * Accepted dialects: tfl, builtin, func
 * Non-Converted Ops: 1, Total Ops 6, % non-converted = 16.67 %
 * 1 ARITH ops

- arith.constant:    1 occurrences  (f32: 1)



  (f32: 1)
```

Note: running the code directly in Jupyter Notebook won't print anything

### Problem

- Why can it not convert all the operations in such a simple model?
- Was it supposed to be like that?
- Does having non-converted operations affect the performance of the model when deployed on a microcontroller using TFLM?
- Is there a way to solve it?

**Comment by gaikwadrahul8** [comment]
This issue originally reported by @Black3rror has been moved to this dedicated repository for [LiteRT](https://github.com/google-ai-edge/LiteRT/) to enhance issue tracking and prioritization. To ensure continuity, we have created this new issue on your behalf.

We appreciate your understanding and look forward to your continued involvement.

**Comment by reid-snc** [comment]
Has there been any movement on this? We wanted to try upgrading to 2.16 or even LiteRT soon for the new features but tflite conversion just seems completely broken..

**Comment by pkgoogle** [comment]
Hi @reid-snc, have you tried with [AI-Edge-Torch](https://github.com/google-ai-edge/ai-edge-torch) instead?

**Comment by jonGuti13** [comment]
Hi! Any update on this? How should we interpret the message? I am converting my keras models (TensorFlow version: 2.15.0) and I face the same warnings.

**Comment by pkgoogle** [comment]
Hi @jonGuti13, I think ignoring those errors/warnings should be fine, do you see any performance issues with accuracy or anything else? Also it's possible just updating keras/TF will work well. If not, in general we do suggest switching over to [AI-Edge-Torch](https://github.com/google-ai-edge/ai-edge-torch) if possible.

**Comment by jonGuti13** [comment]
Hi @pkgoogle,

Just to give you a bit of context,  I’m trying to speed up the preprocessing stage of my network by comparing how different layers handle the same operation. For example, cropping can be done using a crop layer, symmetric convolutions, or asymmetric convolutions, and I want to find out which approach is the fastest.

Since I’m deploying on embedded hardware, I’m relying on XNNPack and ARM NN delegates. When I run the conversion, I get some warnings, and I’m not sure if they affect the results or if there’s something I should adjust to resolve them.

Regarding switching to AI-Edge-Torch, I’d also like to know whether the conversion of 2D convolutions with both stride > 1 and dilation > 1 is supported. I know that TensorFlow 2 doesn’t allow this combination, while PyTorch does, but since TFLite was originally designed for TensorFlow, I’m not sure if this is supported in your stack.

Thanks a lot for your help!

**Comment by pkgoogle** [comment]
Hi @jonGuti13, The vast majority of times, these warnings won't affect the accuracy/precision (of course if you are able to prove differently, we would love to take a look -- though we also prefer in a more up to date environment if possible). Thank you for the additional context, it usually helps us helps you more 😄 . AFAIK, PyTorch supports both those things and the 2d Conv kernel also does as well: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/conv.cc. I would say at this point LiteRT/TFLite is non-edge ML framework agnostic, so I wouldn't worry about non-support for PyTorch workflows.

**Comment by jonGuti13** [comment]
@pkgoogle Thank you again! You were right and with the 2.19 version of TF2 there are no warnings at all.

Speaking about the dilated and strided convolution, you were also right and I can used it in LiteRT/TFLite after converting the Pytorch model with the tool you suggested to me. I thought that LiteRT/TFLite was still closely tied to TensorFlow but, as you point out, it seems to be ML framework agnostic.

**Comment by pkgoogle** [comment]
Thanks @jonGuti13! It appears this issue is resolved, please feel free to close the issue if you have no more open items.

**Comment by jonGuti13** [comment]
@pkgoogle It is @gaikwadrahul8 who is the only one that can close it.
